### ISSUE TITLE
background job runner and multiple hosts?

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-flow#4922]
This is currently legal:

```ini
[platforms]
   [[bg-jobs]]
        hosts = foo, bar, baz  # OOPS
        job runner = background
```

However it doesn't make sense for the background job runner to allow multiple interchangeable hosts, because background jobs have to be polled and killed by PID on the exact host they were submitted to. (Similarly `at`, presumably).

So I guess we should either:
1. make it an error to list multiple hosts for the background job runner
2. or handle background jobs differently: allow multiple job submission hosts but always interact with jobs via the specific host they were submitted to

(And either way, document that background platforms are different in this respect).

What do you think @wxtim ?  (I don't think we're already doing 2. above?)

**Comment by dpmatthews** [comment]
> make it an error to list multiple hosts for the background job runner

Seems reasonable (and for `at`).
Not really a bug though - it's really just a configuration issue that we can catch (there are plenty we can't).

> or handle background jobs differently: allow multiple job submission hosts but always interact with jobs via the specific host they were submitted to

No - this is what platform groups are for.

**Comment by hjoliver** [comment]
> > make it an error to list multiple hosts for the background job runner
> 
> Seems reasonable (and for `at`). Not really a bug though - it's really just a configuration issue that we can catch (there are plenty we can't).

Fair enough. It is a likely configuration mistake though - it would be easy to think a background platform should list of "all the hosts
we can run background jobs on".
 
> 
> > or handle background jobs differently: allow multiple job submission hosts but always interact with jobs via the specific host they were submitted to
> 
> No - this is what platform groups are for.

:+1:

**Comment by oliver-sanders** [comment]
> make it an error to list multiple hosts for the background job runner

Makes sense to me.
